compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1619,49 +1619,48 @@ setTransposeLikeOpRootConfig(func::FuncOp entryPointFn,
     return failure();
   }
 
-  unsigned numLoops = genericOp.getNumLoops();
   DistributionHeuristicConfig distConfig;
   distConfig.minTileSizes = getMinTilingSizesForEachDim(
       entryPointFn, genericOp, linalgOpInfo, targetMLTransInfo);
-  if (llvm::all_of(distConfig.minTileSizes,
-                   [](int64_t vs) { return vs == 1; })) {
-    // Nothing to vectorize just lower to loops.
-    return failure();
-  }
-
   if (llvm::count_if(distConfig.minTileSizes,
                      [](int64_t tileSize) { return tileSize > 1; }) != 2) {
     // Transpose patterns are not applicable if vectorizing more or less than
     // two dims.
     return failure();
   }
 
-  // Make sure that the original tile sizes are multiple of the tile sizes
-  // to be used for the transpose op (i.e., 8x8).
-  // TODO(diegocaballero): Enable 4x8 tile sizes if we find it useful.
-  if (llvm::any_of(distConfig.minTileSizes, [](int64_t tileSize) {
-        return tileSize > 1 && (tileSize % 8) != 0;
+  // Make sure that the original tile sizes are greater than or equal to the
+  // tile sizes to be used for the transpose op (e.g., 8x8, 16x16, etc).
+  int64_t targetVectorSize = 8;
```

Comment:
Do we want this to be arch dependent?

Reply:
We checked the arch in the entry. If it does not have AVX2 features, we bail out. So 8 is a reasonable default. Then the next snippet is that we are evaluating if 16 makes sense.